hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/TokenUpdateLogic.java

```diff
@@ -73,11 +75,37 @@ public TokenUpdateLogic(
     }
 
     public void updateToken(TokenUpdateTransactionBody op, long now) {
+        updateToken(op, now, false);
+    }
+
+    /**
+     * Given a token update transaction and the current consensus time, updates the token in the store.
+     *
+     * <p>The third {@code mergeUnsetMemoFromExisting} argument says whether to preserve the token's
+     * existing memo if the transaction memo is empty. We need this because the current application
+     * binary interface (ABI) for the {@code tokenUpdate()} system contract does not let us distinguish
+     * between a contract omitting the memo in an token update vs. a contract explicitly setting the
+     * memo to the empty string.
+     *
+     * <p>Once the ABI is improved to let a contract advertise it truly does want to erase a memo,
+     * we can remove the {@code mergeUnsetMemoFromExisting} argument and just use the protobuf
+     * message's {@code hasMemo()} method.
+     *
+     * @param op the token update transaction
+     * @param now the current consensus time
+     * @param mergeUnsetMemoFromExisting whether to preserve the token's memo if the transaction memo is unset
+     */
+    public void updateToken(TokenUpdateTransactionBody op, long now, boolean mergeUnsetMemoFromExisting) {
         final var tokenID = tokenValidityCheck(op);
         if (op.hasExpiry()) {
             validateTrueOrRevert(validator.isValidExpiry(op.getExpiry()), INVALID_EXPIRATION_TIME);
         }
         MerkleToken token = tokenStore.get(tokenID);
+        final var isOpMemoUnset = !op.hasMemo() || op.getMemo().getValue().length() == 0;
```

Comment:
Why do we need this condition `op.getMemo().getValue().length() == 0` . Shouldn't `!op.hasMemo()` be enough ?